### PR TITLE
Charles android tasks

### PR DESCRIPTION
**Pull request by charlesx2013**

### CODE REVIEW

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskDetailFragment.java [hunk 1]
```diff
@@ -126,11 +125,12 @@ private void initializeViews(View view) {
 
         if (mTask.getColor() != null) {
             mParentActivity.setToolBarColor(Color.parseColor(mTask.getColor()),
-                    Color.parseColor(Utility.getSecondaryColor(mParentActivity, mTask.getColor())));
+                                            Utility.getSecondaryColor(mParentActivity, mTask.getColor()));
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskDetailFragment.java]: What happens to the color of the task if mTask.getColor() is null

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Utilities/Utility.java [hunk 2]
```diff
@@ -123,10 +122,14 @@ public static void showAndBuildDialog(Activity activity, String title, String me
...
-        String[] COLORS = context.getResources().getStringArray(R.array.colors);
-        String[] SECONDARY_COLORS = context.getResources().getStringArray(R.array.secondary_colors);
-        return SECONDARY_COLORS[Arrays.asList(COLORS).indexOf(primary)];
+    public static int getSecondaryColor(Context context, String primary) {
+        if (primary == null) return Color.parseColor("#81B4DE");
+        int[] COLORS = context.getResources().getIntArray(R.array.colors);
+        int[] SECONDARY_COLORS = context.getResources().getIntArray(R.array.secondary_colors);
+        int color = Color.parseColor(primary);
+        int foundColor = -1;
+        for (int i = 0; i < COLORS.length; i++) { if (COLORS[i] == color) foundColor = i; }
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Utilities/Utility.java]: can we break up these if statements into multiple lines? (same with for loop)

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskDetailFragment.java (hunk 1 above)
Comment (charlesx2013) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskDetailFragment.java]: set to default

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Utilities/Utility.java [hunk 3]
```diff
@@ -123,10 +122,18 @@ public static void showAndBuildDialog(Activity activity, String title, String me
     }
 
 
-    public static String getSecondaryColor(Context context, String primary) {
-        if (primary == null) return "#1976D2";
-        String[] COLORS = context.getResources().getStringArray(R.array.colors);
-        String[] SECONDARY_COLORS = context.getResources().getStringArray(R.array.secondary_colors);
-        return SECONDARY_COLORS[Arrays.asList(COLORS).indexOf(primary)];
+    public static int getSecondaryColor(Context context, String primary) {
+        if (primary == null) return Color.parseColor("#81B4DE");
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Utilities/Utility.java]: should also bracket this if statement

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Utilities/Utility.java [hunk 4]
```diff
@@ -123,10 +122,18 @@ public static void showAndBuildDialog(Activity activity, String title, String me
...
+        int[] COLORS = context.getResources().getIntArray(R.array.colors);
+        int[] SECONDARY_COLORS = context.getResources().getIntArray(R.array.secondary_colors);
+        int color = Color.parseColor(primary);
+        int foundColor = -1;
+
+        for (int i = 0; i < COLORS.length; i++) {
+            if (COLORS[i] == color) foundColor = i;
+        }
+
+        if (foundColor == -1) return Color.parseColor("#81B4DE");
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Utilities/Utility.java]: and this one

**Pull request merged**